tests/posix/xsi_realtime/src/main.c

```diff
@@ -4,6 +4,59 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <pthread.h>
+
 #include <zephyr/ztest.h>
 
+ZTEST(xsi_realtime, test_sched_getparam)
+{
+	struct sched_param param;
+	int rc = sched_getparam(0, &param);
+	int err = errno;
+
+	zassert_true((rc == -1 && err == ENOSYS));
```

Comment:
while you are here, can we fix this and the following?
```suggestion
	zassert_true((rc == -1) && (err == ENOSYS));
```